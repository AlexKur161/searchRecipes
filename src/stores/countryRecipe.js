import { defineStore } from "pinia";
import { getRecipeCountry } from "../axios/index.js";
import { ref, computed, reactive } from "vue";

export const useCountryRecipe = defineStore("country", () => {
  const recipesCountry = ref([]);

  async function countryAction(country) {
    console.log("test1", country);
    const response = await getRecipeCountry(country);
    recipesCountry.value = response;
    console.log(response);
  }

  return { countryAction, recipesCountry };
});
