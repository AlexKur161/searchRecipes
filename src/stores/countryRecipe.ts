import { defineStore } from "pinia";
import { getRecipeCountry } from "../services/recipeCountry";
import { ref, type Ref } from "vue";
import { Card } from "src/types";

export const useCountryRecipe = defineStore("country", () => {
  const recipesCountry: Ref<Card[]> = ref([]);
  const visible: Ref<boolean> = ref(false);

  async function countryAction(country: string) {
    visible.value = true;
    const response = await getRecipeCountry(country);
    response.forEach(
      (item: Card) => (item.id = globalThis.crypto.randomUUID())
    );
    recipesCountry.value = response;
    visible.value = false;
  }

  return { countryAction, recipesCountry, visible };
});
