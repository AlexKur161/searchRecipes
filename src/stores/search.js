import { defineStore } from "pinia";
import { getRecipe } from "../axios/index.js";
import { ref, computed, reactive } from "vue";

export const useSearchRecipe = defineStore("search", () => {
  const recipes = ref([]);

  async function searchAction(name, diet, health, meal) {
    console.log("test1", diet);
    const response = await getRecipe(name, diet, health, meal);
    recipes.value = response;
    console.log(response);
  }

  return { searchAction, recipes };
});
