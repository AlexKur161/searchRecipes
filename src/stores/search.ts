import { defineStore } from "pinia";
import { getRecipe } from "../services/recipe";
import { ref } from "vue";
import { Card } from "src/types/index.ts";

export const useSearchRecipe = defineStore("search", () => {
  const recipes = ref<Card[]>([]);

  const visible = ref<boolean>(false);

  async function searchAction(
    name: string,
    diet?: string,
    health?: string,
    meal?: string
  ) {
    visible.value = true;
    const response = await getRecipe(name, diet, health, meal);
    response.hits.forEach(
      (item: Card) => (item.id = globalThis.crypto.randomUUID())
    );
    recipes.value = response.hits;
    visible.value = false;
  }
  return { searchAction, recipes, visible };
});
