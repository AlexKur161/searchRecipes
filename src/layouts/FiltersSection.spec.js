import { describe, expect, it, beforeEach, vi } from "vitest";
import { mount, config, flushPromises } from "@vue/test-utils";
import component from "./FiltersSection.vue";
import { setActivePinia, createPinia } from "pinia";
import { useSearchRecipe } from "../stores/search";
import { createI18n } from 'vue-i18n'

describe("FiltersSection", () => {
  beforeEach(()=> {
    setActivePinia(createPinia());
  })
  it("", async () => {
    const store = useSearchRecipe();
    const i18n = createI18n({
      legacy: false
    })
    console.log('store.recipes', store.recipes)
    const wrapper = mount(component, {
      global: {
        plugins: [i18n]
      }
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
