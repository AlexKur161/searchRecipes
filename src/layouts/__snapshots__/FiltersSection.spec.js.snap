// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FiltersSection >  1`] = `
"<div data-v-55d9bde9="" id="search" class="section-block">
  <div data-v-55d9bde9="" class="container">
    <h2 data-v-55d9bde9="" class="text-white text-center filter-title">Recipes for every taste</h2>
    <div data-v-5703595a="" data-v-55d9bde9="" class="filter-wrap row">
      <q-input data-v-5703595a="" rounded="" class="select-filter" color="teal-8" bg-color="white" style="max-width: 400px;" modelvalue="" label="The name of the recipe"></q-input>
      <q-select data-v-5703595a="" class="select-filter" color="teal-8" modelvalue="" options="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" emit-value="" map-options="" bg-color="white" label="Diet Labels" transition-show="scale" transition-hide="scale"></q-select>
      <q-select data-v-5703595a="" class="select-filter" color="teal-8" modelvalue="" options="[object Object],[object Object],[object Object],[object Object],[object Object]" emit-value="" map-options="" bg-color="white" label="Meal time" transition-show="scale" transition-hide="scale"></q-select>
      <q-select data-v-5703595a="" class="select-filter" color="teal-8" modelvalue="" options="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" emit-value="" map-options="" bg-color="white" label="Health" transition-show="scale" transition-hide="scale"></q-select>
      <q-btn data-v-5703595a="" class="settings-icon" dense="" round="" size="20px" flat="" icon="tune" color="teal-8"></q-btn>
      <q-btn data-v-5703595a="" class="search-btn" color="teal-8" label="Search"></q-btn>
    </div>
    <div data-v-55d9bde9="" class="nodata-wrapper">Select the parameters to filter</div>
  </div>
</div>"
`;
